### PR TITLE
In which .prof states and events are written

### PR DESCRIPTION
**Pull request by mturilli**
As discussed, this add information about in which .prof file and in which sandbox (client or agent) each state and event is written.

NOTE: events can be configuration dependent so not all of them are recorded in each session. I run a test session with default configuration and added information for all the events that were recorded. I left the others 'blank'.

### CODE REVIEW

File: docs/source/internals.rst [hunk 1]
```diff
@@ -287,30 +285,29 @@ AgentExecutingComponent: (Component)
...
+   app_stop            : application executable stops               (uid: task, [APP])           -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
+   rank_stop           : task exec script: executable stopped       (uid: task)                  -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
+   exec_post           : task exec script: post-exec starts         (uid: task)                  -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
+   exec_stop           : task exec script: stopped                  (uid: task)                  -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
+   launch_collect      : task launch script: launch method returned (uid: task)                  -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
+   launch_post         : task launch script: post-submission        (uid: task)                  -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
+   launch_stop         : task launch script: completes              (uid: task)                  -- agent (task.<TASK_NUMBER>.prof, agent_staging_output.0000.prof)
```
Comment (mtitov) [inline docs/source/internals.rst:307]: no `agent_staging_output.0000.prof`, only in tasks' profs

File: docs/source/internals.rst [hunk 2]
```diff
@@ -169,9 +168,9 @@ Pilot (in session profile, all optional)
 
 ::
 
-   staging_in_start    : pilot level staging request starts         (uid: pilot, msg: did, [PILOT-DS])
-   staging_in_fail     : pilot level staging request failed         (uid: pilot, msg: did, [PILOT-DS])
-   staging_in_stop     : pilot level staging request stops          (uid: pilot, msg: did, [PILOT-DS])
+   staging_in_start    : pilot level staging request starts         (uid: pilot, msg: did, [PILOT-DS]) -- agent; client (rp.session.<SID>.prof)
+   staging_in_fail     : pilot level staging request failed         (uid: pilot, msg: did, [PILOT-DS]) -- agent; client (rp.session.<SID>.prof)
+   staging_in_stop     : pilot level staging request stops          (uid: pilot, msg: did, [PILOT-DS]) -- agent; client (rp.session.<SID>.prof)
```
Comment (mtitov) [inline docs/source/internals.rst:173]: `<SID>` would be a full name including `rp.session`

File: docs/source/internals.rst [hunk 3]
```diff
@@ -131,11 +130,11 @@ Session (Component)
 
 ::
 
-   session_start       : session is being created (not reconnected) (uid: sid)
-   session_close       : session close is requested                 (uid: sid)
-   session_stop        : session is closed                          (uid: sid)
-   session_fetch_start : start fetching logs/profs/json after close (uid: sid, [API])
-   session_fetch_stop  : stops fetching logs/profs/json after close (uid: sid, [API])
+   session_start       : session is being created (not reconnected) (uid: sid)        -- agent; client (rp.session.<SID>.prof)
```
Comment (mtitov) [inline docs/source/internals.rst:133]: `<SID>.prof`

File: docs/source/internals.rst (hunk 2 above)
Comment (mturilli) [reply docs/source/internals.rst]: Right. Went with `*` so not to have to explain what a SID is.

**Review by mtitov — APPROVED** [review]
LGTM!

**Pull request merged**